hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenDissociateFromAccountHandler.java

```diff
@@ -45,13 +70,192 @@ public TokenDissociateFromAccountHandler() {
-        throw new UnsupportedOperationException("Not implemented");
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final TokenDissociateTransactionBody op = txn.tokenDissociateOrThrow();
+
+        validateTruePreCheck(op.hasAccount(), INVALID_ACCOUNT_ID);
+
+        validateTruePreCheck(!TokenListChecks.repeatsItself(op.tokensOrThrow()), TOKEN_ID_REPEATED_IN_TOKEN_LIST);
+    }
+
+    /**
+     * Performs checks that require state and context
+     */
+    @NonNull
+    private ValidatedResult validateSemantics(
+            @NonNull final AccountID accountId,
+            @NonNull final List<TokenID> tokenIds,
+            @NonNull final WritableAccountStore accountStore,
+            @NonNull final ReadableTokenStore tokenStore,
+            @NonNull final WritableTokenRelationStore tokenRelStore,
+            @NonNull final ExpiryValidator expiryValidator) {
+        // Check that the account is usable
+        final var acct = TokenHandlerHelper.getIfUsable(accountId, accountStore, expiryValidator);
+
+        // Construct the dissociation for each token ID
+        final var dissociations = new ArrayList<Dissociation>();
+        for (final var tokenId : tokenIds) {
+            final var tokenRel = tokenRelStore.get(accountId, tokenId).orElse(null);
+            validateTrue(tokenRel != null, TOKEN_NOT_ASSOCIATED_TO_ACCOUNT);
+
+            // Here we check/retrieve a token that may not be "usable," but since we are dissociating token relations,
+            // we don't require a usable token (or even an existing token). We only need to update the token relation
+            final var possiblyUnusableToken = tokenStore.get(tokenId);
+            final TokenRelation dissociatedTokenTreasuryRel;
+            if (possiblyUnusableToken != null) {
+                validateFalse(possiblyUnusableToken.paused(), TOKEN_IS_PAUSED);
+                final var tokenTreasuryAcct = AccountID.newBuilder()
+                        .accountNum(possiblyUnusableToken.treasuryAccountNumber())
+                        .build();
+                dissociatedTokenTreasuryRel =
+                        tokenRelStore.get(tokenTreasuryAcct, tokenId).orElse(null);
+            } else {
+                // If the token isn't found, assume the treasury token rel is null
+                dissociatedTokenTreasuryRel = null;
+            }
+
+            final Dissociation dissociation =
+                    new Dissociation(possiblyUnusableToken, tokenRel, acct, dissociatedTokenTreasuryRel);
+            dissociations.add(dissociation);
+        }
+
+        return new ValidatedResult(acct, dissociations);
     }
+
+    private boolean tokenIsExpired(final Token token, final Instant consensusNow) {
+        // just to get the compiler not to complain about the method params not being used...
+        log.info("tokenIsExpired | token: {}, consensusNow: {}", token, consensusNow);
```

Comment:
This may be a bit noisy, maybe just add a `requireNonNull()` for each parameter or something?

Reply:
We can't require `token` to be non-null since it may have been removed, but I'll just let the compiler complain

Reply:
Done